### DISCUSSION TITLE
Scale down to 0

### DISCUSSION THREAD

**Discussion by Ulrar** [cloudnative-pg/cloudnative-pg#4176] [Q&A] [answered]
Hi,

It looks like the cluster spec requires at least 1 pod. Unfortunately my filesystem is in a bad state and I need to run a repair on it, but I can't figure out a way to free it up as whatever I do, the cluster schedules a pod and the PVC is in use. I've tried adding a nodeSelector to prevent the pod from being scheduled, but that seems to be ignored.

Is there a way to stop a cluster and let the node unmount the pvc so I can run an e2fsck on it ?
I don't want to actually delete the cluster or the PVC will get deleted with it, and recovering from a backup would be a much longer operation.

Thanks

**Comment by Ulrar** [comment]
I ended up going with this : 
```
kubectl taint nodes <all of my nodes> myTaint=blah:NoSchedule
```
Then kill the pod, and because everything is tainted it can't be scheduled. That works but obviously you can't do that on a production cluster, fine for my home cluster though

**Comment by gbartolini** [comment, marked as answer]
You might want to look at [declarative hibernation](https://cloudnative-pg.io/documentation/current/declarative_hibernation/).

**Comment by gilad-aperio** [comment]
Thank you @gbartolini , the hibernation feature is exactly what I needed.

Out of curiosity, what is the reason for having a dedicated feature for this instead of applying the same flow when replicas are scaled to zero? There are other tools ([example](https://github.com/caas-team/gokubedownscaler)) that use the scale subresource to automatically suspend/resume workloads.
